TW-918 Live Collections (#4647)
postmanlabs/postman-docs: src/pages/docs/collections/live-collections/update-live-collections.md

---

Whenever Live Collections observes a new API event in the service that was instrumented, it will trigger an update workflow. Updates can apply to existing requests or result in new requests being added to the collection. Any updated params or headers won't be enabled by default. You can choose to either automatically update the collection or to manually review updates before making them.

## Contents

* [Automatic updates to a Live Collection](#automatic-updates-to-a-live-collection)
* [Manual updates to a Live Collection](#manual-updates-to-a-live-collection)
* [Viewing changes in a Live Collection](#viewing-changes-in-a-live-collection)

## Automatic updates to a Live Collection

After you first create a Live Collection, when the first request is detected, you'll see a notification next to the collection. By default, the request will automatically be added to the collection. To automatically update a Live Collection as it detects updates, select **Automatically update collection**, and then select **Save**.

<img alt="Live Collections update request" src="https://assets.postman.com/postman-docs/v10/live-collections-update-request-first-time-popup.jpg" />

> Selecting automatic updates will result in requests getting automatically created or updated any time the API events are observed in the service.

Reviewer: can we add that no notifications will be received for this - "Selecting automatic updates will result in requests getting automatically created or updated any time the API events are observed in the service without any notifications being sent"

To see a log of updates made automatically, you can use the collection change log. See [Viewing changes in a Live Collection](#viewing-changes-in-a-live-collection) for more

## Manual updates to a Live Collection

You can also choose to review changes and apply them manually, instead of using automatic updates. Every time a new update comes in, you will be notified by the Live Collections icon <img alt="Live Collections icon" src="https://assets.postman.com/postman-docs/v10/icon-live-collections.jpg#icon" width="16px"> next to the collection name turning solid orange. You can then choose to manually update the Live Collection.

To manually update a Live Collection, do the following:

> 1. Hover over the Live Collections icon <img alt="Live Collections icon" src="https://assets.postman.com/postman-docs/v10/icon-live-collections.jpg#icon" width="16px"> and select **Review changes**.

Reviewer: Can we also call out the other entry point from the settings page to set the option to manually review changes

    <img alt="Live Collections update request" src="https://assets.postman.com/postman-docs/v10/live-collections-review-changes-popup.jpg" />

> 1. Review the incoming changes. These can be an addition of a new request or changes to an existing request. Postman also creates new examples based on the response received from the API traffic.

Reviewer: We wanted to remove mentions of traffic. Can we change this to "examples based on the response received from the API events on the application"

---
title: "Updating Live Collections"
updated: 2023-03-15
---

> Whenever Live Collections detect a new event from the API traffic, it will trigger an update workflow. Updates can apply to existing requests or result in new requests being added to the collection. Any updated params or headers won't be enabled by default. You can choose to either automatically update the collection or to manually review updates before making them.

Reviewer 2: @aravind-arun Aren't we avoiding Traffic terminology?

Reviewer: Yes. I propose changing it to 'Whenever Live Collections observes a new API event in the service that was instrumented'

## Contents

* [Automatic updates to a Live Collection](#automatic-updates-to-a-live-collection)
* [Manual updates to a Live Collection](#manual-updates-to-a-live-collection)
* [Viewing changes in a Live Collection](#viewing-changes-in-a-live-collection)

## Automatic updates to a Live Collection

> When the first request is detected by the Live Collection, you'll see a notification next to the collection. By default, the request will automatically be added to the collection. You can also choose to either automatically update the collection or to manually review updates before making them.

Reviewer 3: It seems like this section is about selecting the **Automatically update collection** radio button. What if we update this to something like:

`When the first request is detected by the Live Collection, you'll see a notification next to the collection. By default, the request will automatically be added to the collection. To automatically update a Live collection as it detects updates, select **Automatically update collection**, and then select **Save**.`

The content in this sentence might be a good candidate for the intro: `You can also choose to either automatically update the collection or to manually review updates before making them.`

> When the first request is detected by the Live Collection, you'll see a notification next to the collection. By default, the request will automatically be added to the collection. To automatically update a Live collection as it detects updates, select **Automatically update collection**, and then select **Save**.

Reviewer 2: > you'll see a notification next to the collection
It won't happen for all the collections. It only happens once in the onboarding workflow.

<img alt="Live Collections update request" src="https://assets.postman.com/postman-docs/v10/live-collections-update-request-first-time-popup.jpg" />

> Selecting automatic updates will result in requests getting automatically created or updated anytime they are observed in the API traffic.

Reviewer 2: Same comment as the above related to Traffic terminology.

Reviewer: We can change it to "Selecting automatic updates will result in requests getting automatically created or updated anytime the API events are observed in the service"

> Every time a new update comes in, you will be notified by the live Collections icon <img alt="Live collections icon" src="https://assets.postman.com/postman-docs/v10/icon-live-collections.jpg#icon" width="16px"> next to the collection name turning orange.

Reviewer 4: ```suggestion
Every time the collection is updated, you will be notified by the Live Collections icon <img alt="Live collections icon" src="https://assets.postman.com/postman-docs/v10/icon-live-collections.jpg#icon" width="16px"> next to the collection name turning solid orange.
```

> Every time the collection is updated, you will be notified by the Live Collections icon <img alt="Live collections icon" src="https://assets.postman.com/postman-docs/v10/icon-live-collections.jpg#icon" width="16px"> next to the collection name turning solid orange.

Reviewer 2: This doesn't happen for auto updates. User can check the updates on the Changelogs but there is no UI indication for auto updates.

Reviewer: @jkonrath-postman - Can we follow a structure similar to automatic updates section mentioned here - https://postmanlabs.atlassian.net/wiki/spaces/AO/pages/4404053133/Automatic+Updates+to+the+Postman+Live+Collection
"Whenever the first request is detected by the live collection, it will automatically add it to the collection. This will also provide a choice to either automatically update the collection or to manually review before updating. Setting automatic updates will result in requests getting automatically created or updated anytime they are observed in the API traffic without any user notification"

## Manual updates to a Live Collection

You can also choose to review changes and apply them manually, instead of using automatic updates.

To manually update a Live Collection, do the following:

> 1. Select manual updates from the first request popup shown above, or in [Live Collections settings](/docs/collections/live-collections/live-collections-settings).

Reviewer 3: To align with my above suggestion, what if we update this to something like:

`1. In the notification next to the collection, select **Review before updating**, and then select **Save**. You can also do this in [Live Collections settings](/docs/collections/live-collections/live-collections-settings).

> 1. In the notification next to the collection, select **Review before updating**, and then select **Save**. You can also do this in [Live Collections settings](/docs/collections/live-collections/live-collections-settings).

Reviewer 2: We haven't explained what this notification refers to. We should add some info around that.

Reviewer: @jkonrath-postman - By notification here, do you mean the following notification that was called out in the automatic updates section? "When the first request is detected by the Live Collection, you'll see a notification next to the collection..."

Author: I just meant that the lightning bolt turns from hollow and gray to full and orange. (Although we shouldn't say orange because we try not to refer to color in the docs because of accessibility.

> 1. When the live collections icon <img alt="Live collections icon" src="https://assets.postman.com/postman-docs/v10/icon-live-collections.jpg#icon" width="16px"> next to the collection name is orange, select it, then select **Review changes**.

Reviewer 3: Instead of "orange", what if we wrote something like "solid". I'm suggesting this because I'm not sure how using a color name might impact folks who are colorblind.

> 1. When the Live Collections icon <img alt="Live Collections icon" src="https://assets.postman.com/postman-docs/v10/icon-live-collections.jpg#icon" width="16px"> next to the collection name is orange, select it, then select **Review changes**.

Reviewer 4: ```suggestion
1. When the Live Collections icon <img alt="Live Collections icon" src="https://assets.postman.com/postman-docs/v10/icon-live-collections.jpg#icon" width="16px"> next to the collection name is solid orange, select it, then select **Review changes**.
```

    <img alt="Live Collections update request" src="https://assets.postman.com/postman-docs/v10/live-collections-review-changes-popup.jpg" />

1. Review the incoming changes. These can be an addition of a new request or changes to an existing request. Postman also creates new examples based on the response received from the API traffic.

    <img alt="Live Collections review changes" src="https://assets.postman.com/postman-docs/v10/live-collections-review-changes.jpg" />

> 1. Select **Update Collection** to accept the changes and update the collection, or **Reject all** to ignore them.

Reviewer 3: ```suggestion
1. Select **Update Collection** to accept the changes and update the collection, or select **Reject all** to ignore them.
```

## Viewing changes in a Live Collection

To see changes made to a Live Collection, use the collection changelog. The changelog for the collection shows the updates that have been made to a Live Collection.

> To see the changelog, open the collection and select the changelog icon <img alt="Changelog icon" src="https://assets.postman.com/postman-docs/icon-changelog-v9.jpg#icon" width="18px"> on the right.

Reviewer 3: ```suggestion
To see the changelog, open the collection and select the changelog icon <img alt="Changelog icon" src="https://assets.postman.com/postman-docs/icon-changelog-v9.jpg#icon" width="18px"> in the right sidebar.
```

Reviewer 3: This suggestion is based on instructions at `/docs/collections/live-collections/live-collections-sdk-install/`. When we reference the live collections icon, we mention that it's in the right sidebar.

> To see the changelog, open the collection and select the changelog icon <img alt="Changelog icon" src="https://assets.postman.com/postman-docs/icon-changelog-v9.jpg#icon" width="18px"> in the right sidebar.

Reviewer 2: User can also go to the Changelogs from the hover over information on top of the lightning / Live Collection icon.

Reviewer: @jkonrath-postman - The lightning icon next to the collection name can also serve as an entry point for 2 workflows
1. Viewing the change log
2. Live settings page
Lets add this as an entry point in both these places
<img width="781" alt="Screen Shot 2023-03-14 at 1 07 28 AM" src="https://user-images.githubusercontent.com/119476416/224936020-9952f851-2ae8-4749-9419-3076c6a0e21e.png">

> <img alt="Changelog view" src="https://assets.postman.com/postman-docs/v10/changelog-view.jpg" width="250px"/>

Reviewer 3: What do you think about increasing the images width? It's a little difficult to read the text in the image.
